### DISCUSSION TITLE
How to use module federation runtime with NX

### DISCUSSION THREAD

**Discussion by Wykks** [nrwl/nx#28433] [Questions]
Hello!

Since NX use `@module-federation/enhanced`, how to actually use the Federation Runtime (https://module-federation.io/guide/basic/runtime.html) ?

I want to be able to call `init()` and `loadRemote()`. Or is NX doing all this work magically when we call `import()`?

Additionnaly, currently NX still load  "remote-entry.ts" instead of "mf-manifest.json", and I don't see a way to tell nx to use "mf-manifest.json" instead.

**Comment by Wykks** [comment]
ping @Coly010

**Comment by Coly010** [comment]
No, it's not automatically happening with the `import()`. Import still uses build code.


Nx will continue to default to the `remoteEntry.js` for short time yet, but you can use `mf-manifest.json`.

in your `module-federation.config.ts` file, you can pass an array for a remote that points to the mf-manifest.json file instead:

```ts
export const config: ModuleFederationConfig = {
  name: 'myhost',
  remotes: [
      ['remote1', 'http://localhost:4201/mf-manifest.json']
  ]
}
```

To use the runtime helpers, in your bootstrap file, you can call `init` like:

```ts
init({
  name: 'myhost',
  remotes: [
     {
        name: 'myremote',
        entry: 'http://localhost:4201/mf-manifest.json'
     }
  ]
})
```

Then wherever you would usually use `import('myremote/Module')`, you can replace it with `loadRemote('myremote/Module')`
